g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/v2/formatting/pattern/FormattingPatternAPI.java

```diff
@@ -88,10 +90,26 @@ public APIResponseDTO getI18nPatternWithLanguageAndRegion(
         }
 
         Map<String, Object> patternMap = patternService.getPatternWithLanguageAndRegion(language, region, categories, scopeFilter);
-        if (!(boolean)patternMap.get("isExistPattern")) {
-            return super.handleResponse(APIResponseStatus.INTERNAL_NO_RESOURCE_ERROR, "'No mapping language found for region!");
+
+        int emptyCount = getEmptyCategoryCount(patternMap, categories);
+        if(emptyCount == 0) {
+            return super.handleResponse(APIResponseStatus.OK, patternMap);
+        }else if(emptyCount == categories.size()) {
```

Comment:
Total returned category count may be greater than the requested category count because server will add 'number' category if only 'currencies' is requested.

So this isn't enough to determine if all are failed.

Reply:
That doesn't matter, anyway the response code should be '200', the PR code achieve it.